packages/react-component-library/src/fragments/NotificationSidePanel/index.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useEffect, useRef, useState } from 'react'
+
+import { Popover } from '../../components'
+import { Bell } from '../../icons'
+
+interface NotificationSidePanelProps {
+  className?: string
+  unreadNotification?: boolean
+}
+
+const BAR_WIDTH = 60
+const POPOVER_WIDTH = 335
+
+const NotificationSidePanel: React.FC<NotificationSidePanelProps> = ({
+  children,
+  className = '',
+  unreadNotification,
+}) => {
+  const hasNotifications = children !== null
+  const [showNotifications, setShowNotifications] = useState(false)
```

Comment:
Does it make sense to bundle a lot of this into a custom hook?

Reply:
Not sure how I would

Reply:
You could export documentClick and toggleNotifications from the hook and just consume like so:

```const [state, setter, somethingElse, somethingElse] = useCustomHook(arg1, arg2, …)```

Maybe it isn’t warranted where there isn’t going to be code re-use?

Just trying to start a discussion around custom hooks usage - I did this in the Pagination implementation in order to encapsulate related logic, even though it was tightly coupled to the component.

Reply:
Will take a look and do it as a follow up